xml2rfc/parser.py

```diff
@@ -543,47 +542,47 @@ def parse(self, remove_comments=True, remove_pis=False, quiet=False, strip_cdata
-        context.resolvers.add(caching_resolver)
-
-        # Get hold of the rfc number (if any) in the rfc element, so we can
-        # later resolve the "&rfc.number;" entity.
-        self.rfc_number = None
-        self.format_version = None
-        try:
-            for action, element in context:
-                if element.tag == "rfc":
-                    self.rfc_number = element.attrib.get("number", None)
-                    self.format_version = element.attrib.get("version", None)
-                    break
-        except ValueError as e:
-            if e.message=="I/O operation on closed file":
-                pass
+        with io.BytesIO(text) as file:
+            file.name = self.source
+            context = lxml.etree.iterparse(file,
+                                          dtd_validation=False,
+                                          load_dtd=True,
+                                          attribute_defaults=True,
+                                          no_network=self.no_network,
+                                          remove_comments=remove_comments,
+                                          remove_pis=remove_pis,
+                                          remove_blank_text=True,
+                                          resolve_entities=False,
+                                          strip_cdata=strip_cdata,
+                                          events=("start",),
+                                          tag="rfc",
+                                      )
+            # resolver without knowledge of rfc_number:
+            caching_resolver = CachingResolver(cache_path=self.cache_path,
+                                            library_dirs=self.library_dirs,
+                                            templates_path=self.templates_path,
+                                            source=self.source,
+                                            no_network=self.no_network,
+                                            network_locs=self.network_locs,
+                                            verbose=self.verbose,
+                                            quiet=self.quiet,
+                                            options=self.options,
+                                         )
+            context.resolvers.add(caching_resolver)
+
+            # Get hold of the rfc number (if any) in the rfc element, so we can
+            # later resolve the "&rfc.number;" entity.
+            self.rfc_number = None
+            self.format_version = None
+            try:
+                for action, element in context:
+                    if element.tag == "rfc":
+                        self.rfc_number = element.attrib.get("number", None)
+                        self.format_version = element.attrib.get("version", None)
+                        break
+            except ValueError as e:
+                if e.message=="I/O operation on closed file":
+                    pass
```

Comment:
Are we ignoring _all_ `ValueError`s here? Otherwise, 
```python
else:
    raise e
```
may be needed. Let me know if I'm reading it wrong.

Reply:
Good question. It wasn't a change introduced by this PR.
But I don't think we should catch that error.
I'll remove the `try-except`.

Reply:
Addressed with https://github.com/ietf-tools/xml2rfc/pull/1134/commits/b765594962d504d9d59181c4d69135e526959b50